src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,40 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import { Loader } from '../Loader';
+import { getUser } from '../../api';
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
+
+type Props = {
+  todoId: number;
+  todos: Todo[];
+  selectedTodo: (argument: number) => number | void;
+};
+
+export const TodoModal: React.FC<Props> = ({
+  todoId,
+  todos,
+  selectedTodo,
+}) => {
+  const [user, setUser] = useState<User | null>(null);
+  const [todoOpen, setTodoOpen] = useState(true);
+  const currentTodo = todos.find(todo => todo.id === todoId);
+
+  useEffect(() => {
+    if (currentTodo) {
+      getUser(currentTodo.userId)
+        .then(response => setUser(response));
```

Comment:
Let’s use async await. It gives you a simpler way to work with asynchronous promise-based code